portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+var curSlide = 0; //by default, display first slide
+const totalSlides = 3; //total number of slides
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/*
+ * Advance forward by one slide
+ */
+function forward() {
+  if (curSlide < totalSlides - 1) {
+    updateSlide(curSlide + 1);
+  } else { //curSlide = totalSlides
+    updateSlide(0);
+  }
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+/*
+ * Advance backwards by one slide
+ */
+function back() {
+  if (curSlide == 0) {
+    updateSlide(totalSlides - 1);
+  } else {
+    updateSlide(curSlide - 1);
+  }
 }
+
+function updateSlide(slide) {
+  curSlide = slide;
+  var slideArr = document.getElementsByClassName("slide");
+  var defaultSlide = document.getElementsByClassName("default-slide");
+  defaultSlide[0].style.display = "none";
```

Comment:
Although we know that this will exist by looking at the HTML, it's good practice to check that defaultSlide has an element at [0] before accessing it.